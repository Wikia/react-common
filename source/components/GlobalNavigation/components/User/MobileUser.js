import React from 'react';
import PropTypes from 'prop-types';

import MobileAnon from './MobileAnon';
import MobileLoggedIn from './MobileLoggedIn';

const MobileUser = ({ data, openModal, isOpen, track }) => {
    if (data.user) {
        return (
            <MobileLoggedIn
                user={data.user}
                openModal={openModal}
                isOpen={isOpen}
                track={track}
            />
        );
    }

    if (data.anon) {
        return <MobileAnon data={data.anon} />;
    }

    return null;
};

MobileUser.propTypes = {
    data: PropTypes.shape().isRequired,
    isOpen: PropTypes.bool.isRequired,
    openModal: PropTypes.func.isRequired,
    track: PropTypes.func.isRequired,
};

export default MobileUser;
