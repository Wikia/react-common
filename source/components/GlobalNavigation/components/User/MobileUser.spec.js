import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import MobileUser from './MobileUser';

const defaultProps = {
    model: {
        notifications: {
            header: {
                title: {
                    key: 'some-label',
                },
            },
        },
    },
    openModal: () => null,
    track: () => null,
    isOpen: false,
};

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<MobileUser {...computedProps} />);
}

test('MobileUser renders correctly for user', () => {
    expect(renderComponent({ model: { user: { mockedUser: 'mocked-user-value' } } })).toMatchSnapshot();
});

test('MobileUser renders correctly for anon', () => {
    expect(renderComponent({ model: { anon: { mockedAnon: 'mocked-anon-value' } } })).toMatchSnapshot();
});

test('MobileUser returns null when no data is provided', () => {
    expect(renderComponent()).toMatchSnapshot();
});
