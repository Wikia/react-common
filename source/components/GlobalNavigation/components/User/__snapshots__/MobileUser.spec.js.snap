// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileUser renders correctly for anon 1`] = `
<ComponentWrapper
  model={
    Object {
      "mockedAnon": "mocked-anon-value",
    }
  }
/>
`;

exports[`MobileUser renders correctly for user 1`] = `
<MobileLoggedIn
  header="some-label"
  isOpen={false}
  openModal={[Function]}
  track={[Function]}
  user={
    Object {
      "mockedUser": "mocked-user-value",
    }
  }
/>
`;

exports[`MobileUser returns null when no data is provided 1`] = `""`;
